mysql-test/t/mysqldump_gtid_purged.inc

```diff
@@ -0,0 +1,91 @@
+--source include/have_log_bin.inc
+--source include/have_innodb.inc
+--source include/not_embedded.inc
+--source include/have_debug_sync.inc
+--source include/count_sessions.inc
+
+# start-sql-file/finish-sql-file options of mysqldump available only in debug build
+--source include/have_debug.inc
+
+# Handling of background process
+--source include/not_windows.inc
+
+
+--echo #################################################
+--echo # Testing mysqldump with options: $MYSQLDUMP_OPTS
+--echo #################################################
+
+--let $sleep_sec = 5
+--let $start_sql_file = $MYSQL_TEST_DIR/std_data/mysqldump_gtid_purged_start.sql
+--let $finish_sql_file = $MYSQL_TEST_DIR/std_data/mysqldump_gtid_purged_finish.sql
+
+
+--connect(con1, localhost, root)
+
+--connection default
+CREATE TABLE t1 (a INT) ENGINE = InnoDB;
+INSERT INTO t1 VALUES (0);
+# Capture current gtid_executed
+--let $gtid_executed_start = `SELECT @@GLOBAL.GTID_EXECUTED`
+
+# Start mysqldump in background collecting its PID
+--let $command = $MYSQL_DUMP
+--let $command_opt = $MYSQLDUMP_OPTS test t1 --start-sql-file=$start_sql_file --finish-sql-file=$finish_sql_file
+--let $output_file = $MYSQLTEST_VARDIR/tmp/dump.sql
+--let $pid_file = $MYSQLTEST_VARDIR/tmp/mysqldump.pid
+--source include/start_proc_in_background.inc
+
+# Wait for mysqldump to reach the point just before getting gtid_executed
+SET DEBUG_SYNC = 'now WAIT_FOR before_show_gtid_executed.reached';
+
+
+# Now mysqldump started its job.
+# Following insert will change @@global.gtid_executed, but it should not change the snapshot
+# It may block if mysqldump was started with --lock-all-tables, so we need to do it in async way
+--send INSERT INTO t1 VALUES (1)
+--real_sleep $sleep_sec
```

Comment:
What is the  reason for this `real_sleep`?

Reply:
This 'inc' file is used several times during the test to test mysqldump behavior when it is started with different parameters. Depending on parameters, mysqldump may block all tables with read lock (FLUSH TABLES WITH READ LOCK) which will cause all inserts to block, or it may issue LOCK TABLES FOR BACKUP which will NOT cause inserts to block. (BTW this is also the reason why INSET is done from parallel connection in 'async' way. For cases when inserts are not blocked, it could have been done from default connection) 
During the test we need to be sure if INSERT was fully executed, or it was blocked. So the reason for this sleep is to give 5 seconds time for it. Probably it could be achieved with some debug sync points, but there is no single point for these 2 cases that would signal that what we needed was done (blocked or inserted). 
That's why I decided to keep things simple and put sleep here.

Reply:
OK